pytorch_lightning/utilities/distributed.py

```diff
@@ -202,6 +204,14 @@ def all_gather_ddp_if_available(
     Return:
         A tensor of shape (world_size, batch, ...)
     """
+    if group is None:
+        group = torch.distributed.group.WORLD
+        if group is None:
+            raise MisconfigurationException(
+                "The provided group was None and `torch.distributed.group` isn't available."
+                " Gathering tensor across processes won't be possible."
+            )
```

Comment:
Is this really possible/necessary? IMO this would be catched by the below statement (`if torch.distributed.is_available() and torch.distributed.is_initialized()`) if you move the above statement (`if group is None: group = torch.distributed.group.WORLD` to the if clauses scope.

Reply:
removed !